src/main/java/pl/h2security/registration/RegistrationService.java

```diff
@@ -0,0 +1,54 @@
+package pl.h2security.registration;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestBody;
+import pl.h2security.registration.exceptions.EmailAlreadyExistException;
+import pl.h2security.registration.exceptions.UserAlreadyExistException;
+import pl.h2security.registration.exceptions.UserNotExistException;
+import pl.h2security.repository.UserRepository;
+import pl.h2security.service.UserService;
+import pl.h2security.user.User;
+
+import java.util.Optional;
+
+@Service
+public class RegistrationService {
+
+
+    private UserRepository userRepo;
+    private UserService userService;
+
+
+    @Autowired
+    public RegistrationService(UserRepository userRepo,UserService userService) {
+        this.userRepo = userRepo;
+        this.userService=userService;
+    }
+
+    public void register(User user) {
+        if (emailExist(user.getEmail())) {
+            throw new EmailAlreadyExistException("There is account with that email adress.");
+        } else if (loginExist(user.getUserName())){
+            throw new UserAlreadyExistException("There is account with that user name.");
+        }
+        userService.addWithDefaultRole(user);
+    }
+
+    private boolean emailExist(String email) {
```

Comment:
Method name should be suggesting that it returns boolean. For example: isEmailUsed